### PR TITLE
Stop GridBagSizer empty cells inflating gaps

### PR DESCRIPTION
**Pull request by LegalizeAdulthood**
Track which GridBagSizer rows and columns are occupied while calculating minimum size, and apply the configured empty-cell size only to rows and columns with no shown item.

Add a regression test covering spanned items and genuinely empty rows and columns.

Fixes #3105

### CODE REVIEW

File: src/common/gbsizer.cpp [hunk 1]
```diff
@@ -7,6 +7,7 @@
 // Author:      Robin Dunn
 // Created:     03-Nov-2003
 // Copyright:   (c) Robin Dunn
+// Copyright:   (c) 2026 wxWidgets development team
```
Comment (vadz) [inline src/common/gbsizer.cpp]: Minor, but I don't see any point in adding these lines. Git commit logs are authoritative IMO.

File: src/common/gbsizer.cpp [hunk 2]
```diff
@@ -447,6 +448,9 @@ wxSize wxGridBagSizer::CalcMin()
     m_rowHeights.Empty();
     m_colWidths.Empty();
 
+    wxArrayInt rowHasItem;
+    wxArrayInt colHasItem;
```
Comment (vadz) [inline src/common/gbsizer.cpp:451]: This should be `std::vector<bool>`, right? Or `wxVector<bool>` if you'd like to backport this to 3.2.

File: tests/sizers/gridsizer.cpp
Comment (vadz) [inline tests/sizers/gridsizer.cpp]: To be pedantic, this should list `wxGridBagSizer` too now.

**Review by vadz — COMMENTED** [review]
Thanks, I never use `wxGridBagSizer` myself, but I guess this is indeed the correct behaviour.

I can merge this as is, but I'd like to replace `wxArrayInt` with `vector<bool>`, please let me know if you see any problems with it or if you'd rather do it yourself.

Thanks!

File: src/common/gbsizer.cpp (hunk 1 above)
Comment (LegalizeAdulthood) [reply src/common/gbsizer.cpp]: oops, forgot to clean that out.  will do for this and other PRs

Comment (vadz) [reply src/common/gbsizer.cpp]: No, it's ok, leave them, I'll remove (and already removed a couple of) them when applying, thanks.

File: src/common/gbsizer.cpp (hunk 2 above)
Comment (LegalizeAdulthood) [reply src/common/gbsizer.cpp:451]: I used wxArrayInt here because these are just 0/1 scratch arrays kept in lockstep with m_rowHeights/m_colWidths, which are wxArrayInt already, so this keeps the same container API and avoids std::vector<bool>'s special proxy-reference semantics. A bool wx array is not really better for the same reason, and these arrays are tiny and local to CalcMin(), so saving a few bytes would not justify using the bool specialization here.
